Wine storage (bookmarking functionality)

AnnekeSchuette opened the pull request:
## Value statement

As a **user** 
I **need** to see all my saved wines in a list
so that I can  **access** and **read** their information again or remove them from the list

## Acceptance criteria
- the _WineStoragePage_ renders all bookmarked wines => all wines from the _savedWines_ array
- all bookmark-icons on wine cards have the active state
- wines are removed from list by tapping the bookmark icon 

[https://github.com/AnnekeSchuette/capstone-project/issues/14](url)

marcushohlbein commented on client/src/components/App/App.jsx:
```diff
@@ -1,19 +1,31 @@
 export default function App() {
-  const [savedWines, { handleSaveWine }] = useSaveWine()
-  console.log(savedWines)
+  const [savedWines, toggleSave] = useHandleSave()
```
an own custom hook, great 🎉

marcushohlbein commented on client/src/hooks/useHandleSave.js:
```diff
@@ -0,0 +1,27 @@
+import useLocalStorage from 'hooks/useLocalStorage'
+
+export default function useHandleSave() {
```
is the name "handleSave" the right one? as a third person i would expect that you can save something. but it's more like a favorite toogle, isn't it? maybe _"useToggleFavorites"_ could be an option

marcushohlbein reviewed (commented):
your code is a good one, keep on 👌 add some comments in the code

macfrei commented on client/src/components/App/App.jsx:
```diff
@@ -1,19 +1,31 @@
+          <Route path="/wine-storage">
+            <WineStorage savedWines={savedWines} onSaveToggle={toggleSave} />
+          </Route>
+          <Route path="/">
```
Just in case, I think I'd add an `exact` if the `WineListing` component should only be rendered then. I know it's not necessary as you put it at last route in the switch, but you never know :) just my thoughts though.
```suggestion
          <Route exact path="/">
```

macfrei commented on client/src/components/WineListing/WineListing.jsx:
```diff
@@ -0,0 +1,45 @@
+import WineCard from 'components/WineCard/WineCard'
+import styled from 'styled-components/macro'
+import useLocalStorage from 'hooks/useLocalStorage'
```
Beautiful 😻 But where in this file are you using it? Maybe I am overlooking something but I can't see it. Same goes for `useState`. If you don't use these here, you can get rid of the imports.

AnnekeSchuette replied:
This is a obsolete import I've overseen ... 😄  Thanks for the hint!

macfrei reviewed (commented):
Good job! Added a few comments :)

AnnekeSchuette commented:
> Good job! Added a few comments :)
> 
> As far as I could see it from your PR, your code looks clean and well written. You mentioned that you want/need to change a a bit? What is it that concerns you?

AnnekeSchuette commented:
> Good job! Added a few comments :)
> 
> As far as I could see it from your PR, your code looks clean and well written. You mentioned that you want/need to change a a bit? What is it that concerns you?

@macfrei Thank you for the code review and feedback! I just thought the PR was a little chaotic with all of it's commits and changed files. It was more concern regarding the form instead of a code based one. I thought there might be a good way to clean up pull requests before code reviews, so that the reviewer can concentrate on the important stuff. Maybe there is an easy way I don't know yet to create PRs which start later in branch history? I know git rebase function a little, but don't feel confident using it. But one might just say this is unnecessary and makes the work flow less transparent ...
